### PR TITLE
bank: initial project setup

### PR DESCRIPTION
**Pull request by BorislavAnchev**
Initial project setup for the bank app task.

### CODE REVIEW

**Comment by BorislavAnchev** [comment]
@mgenov, code ready for review.

File: banktask/src/App.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import Header from './components/Header/Header';
+import Transaction from './screens/transaction/Transaction';
+
+const App = () => {
+
+  return (
+    <BrowserRouter>
+      <div className="App">
+        <Header data-test='Header Component'/>
+        <Route exact path='/' render={() => <Transaction />} data-testid='Transaction Page Route' />
```
Comment (mgenov) [inline banktask/src/App.js]: HomePage sounds better than Transaction.

File: banktask/src/components/Header/Header.spec.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { shallow } from 'enzyme';
```
Comment (mgenov) [inline banktask/src/components/Header/Header.spec.js]: Using of react-test-renderer improves test-behaviour capabilities. Enzyme enforces coupling of test and implementation which makes code hard for maintenance.

File: banktask/src/screens/transaction/Transaction.spec.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    let wrapper;
+    beforeEach(() => {
+      const initialState = {
+        accounts: {
+          _u70nyuzcq: {
+            id: '_u70nyuzcq',
+            iban: 'BG12BUIN12341234567891',
+            currency: 'BGN',
+            balance: '5678.00',
+            history: [
```
Comment (mgenov) [inline banktask/src/screens/transaction/Transaction.spec.js]: You can lower the test data a little bit. 

You can consider and different design as when you have 1000 transactions and you go through pages you will retrieve the account information for each page.

File: banktask/src/screens/transaction/Transaction.spec.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    });
+
+    describe('Submit Warning paragraph', () => {
+      it('Should contain a submit warning paragraph', () => {
+        const submitWarning = findByTestAttribute(wrapper, 'submitWarning');
+        expect(submitWarning.length).toBe(1);
+      });
+
+      it('The submit warning should update correctly', () => {
+        const { getByTestId } = render(<Provider store={testStore({})}><MemoryRouter><Transaction /></MemoryRouter></Provider>);
```
Comment (mgenov) [inline banktask/src/screens/transaction/Transaction.spec.js]: You can improve test readability by using render wrapper function:

```jsx
const { getByTestId } = render(<Transaction />, {}) 
```

File: banktask/src/App.js (hunk 1 above)
Comment (BorislavAnchev) [reply banktask/src/App.js]: Changed.

File: banktask/src/components/Header/Header.spec.js (hunk 2 above)
Comment (BorislavAnchev) [reply banktask/src/components/Header/Header.spec.js]: Used `react-testing-library` as it is what you recommended on the other PR and what is recommended on the React website. Although, there may be some specific cases in which react-test-rendered is a better solution, and which I am unaware of, so I keep in mind that you may have meant use _exactly_ `react-test-renderer`.

File: banktask/src/screens/transaction/Transaction.spec.js (hunk 3 above)
Comment (BorislavAnchev) [reply banktask/src/screens/transaction/Transaction.spec.js]: Testing data is removed for now, as it is not required by any functionality at this moment. It was merely missed out last time. However, I will keep that in mind to couple only the needed testing data with the specific testing scenarios when such is needed.

File: banktask/src/screens/transaction/Transaction.spec.js (hunk 4 above)
Comment (BorislavAnchev) [reply banktask/src/screens/transaction/Transaction.spec.js]: Done.

**Comment by BorislavAnchev** [comment]
New changes ready for a review:

- Changed the name of the 'Transaction' component to 'HomeScreen'
- Swapped `Enzyme` for `react-testing-library`
- Removed the testing data as it was not used at the moment.
- Removed the wrapping <Provider> and <MemoryRouter> in the render() method to improve readability.

File: banktask/src/App.js (hunk 1 above)
Comment (BorislavAnchev) [reply banktask/src/App.js]: Changed.

**Comment by BorislavAnchev** [comment]
Added some tests for the transaction type dropdown component, but I'm not sure whether I should be writing tests for it or not.

File: banktask/src/screens/HomePage/HomePage.spec.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+describe('Home page screen component ', () => {
+    
+    afterEach(cleanup);
+
+    it('Should render a Home page component', () => {
+      const { getByTestId } = render(<HomePage/>);
+      expect(getByTestId('Home Page Component')).toBeInTheDocument(); // Equavalent to expect(transactionSection.length).toBe(1);
+    });
+
+    describe('Dropdown component', () => {
```
Comment (mgenov) [inline banktask/src/screens/HomePage/HomePage.spec.js]: Doesn't look like it belongs here. If you are testing some external component, you are doing integration test but I'm not sure that you need this at all.

File: banktask/src/utils/index.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { applyMiddleware, createStore } from 'redux';
+import { rootReducer, middlewares } from '../redux/rootReducer';
+
+export const findByTestAttribute = (component, attribute) => {
+  const wrapper = component.find(`[data-testid='${attribute}']`);
+  return wrapper;
+};
+
+export const testStore = (initialState) => {
```
Comment (mgenov) [inline banktask/src/utils/index.js]: This looks really strange as it doesn't contain anything related to tests.

**Comment by BorislavAnchev** [comment]
New changes ready for a review:

- Removed the utility functions for now as they are not needed at the moment
- Removed the tests for the dropdown component

File: banktask/src/redux/modules/account/actions.spec.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        }
+      }
+    }
+    expect(loadAccounts()).toEqual(expectedAction);
+  });
+});
+
+describe('updateAccount', () => {
+  it('updateAccount should return an object with correct type and payload', () => { 
+    const mockId = '_u70nyuzcq';
```
Comment (mgenov) [inline banktask/src/redux/modules/account/actions.spec.js]: naming with "mock" is misleading as you are doing state testing and you are not mocking nothing actually.

**Pull request merged**